### ISSUE TITLE
Unable to pass methods (static or instance) as closures

### ISSUE THREAD

**Issue by azjgard** [winglang/wing#3250]
### I tried this:

Given this starter code:
```js
bring cloud;

let api = new cloud.Api();

api.get("/", inflight (request: cloud.ApiRequest): cloud.ApiResponse => {
  return cloud.ApiResponse {
    status: 200,
    body: "Hello, world!"
  };
});
```

I moved the controller logic into a static method of a class:
```js
inflight class RequestController {
  static inflight handleRequest(request: cloud.ApiRequest): cloud.ApiResponse {
    return cloud.ApiResponse {
      status: 200,
      body: "Hello, world!"
    };
  }
}
```

And then tried to use that static method to handle requests:
```js
api.get("/", RequestController.handleRequest);
```

### This happened:

The following error was thrown during compilation:
```
ERROR: Cannot read properties of undefined (reading 'node')

target/gdpr.wsim.233847.tmp/.wing/preflight.js:44
       }
       const api = this.node.root.newAbstract("@winglang/sdk.cloud.Api",this,"cloud.Api");
>>     (api.get("/",RequestController.handleRequest));
     }
   }
```

### I expected this:

I expected the console to compile and requests to be handled successfully.

### Is there a workaround?

Don't use class methods for handling API requests, and define route controllers using anonymous inflight functions (like in the original code sample).

### Component

Compiler

### Wing Version

0.23.17

### Node.js Version

v20.3.1

### Platform(s)

MacOS

### Anything else?

- Playground [link](https://play.winglang.io/?code=YgByAGkAbgBnACAAYwBsAG8AdQBkADsACgAKAGwAZQB0ACAAYQBwAGkAIAA9ACAAbgBlAHcAIABjAGwAbwB1AGQALgBBAHAAaQAoACkAOwAKAAoAaQBuAGYAbABpAGcAaAB0ACAAYwBsAGEAcwBzACAAUgBlAHEAdQBlAHMAdABDAG8AbgB0AHIAbwBsAGwAZQByACAAewAKACAAIABzAHQAYQB0AGkAYwAgAGkAbgBmAGwAaQBnAGgAdAAgAGgAYQBuAGQAbABlAFIAZQBxAHUAZQBzAHQAKAByAGUAcQB1AGUAcwB0ADoAIABjAGwAbwB1AGQALgBBAHAAaQBSAGUAcQB1AGUAcwB0ACkAOgAgAGMAbABvAHUAZAAuAEEAcABpAFIAZQBzAHAAbwBuAHMAZQAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAYwBsAG8AdQBkAC4AQQBwAGkAUgBlAHMAcABvAG4AcwBlACAAewAKACAAIAAgACAAIAAgAHMAdABhAHQAdQBzADoAIAAyADAAMAAsAAoAIAAgACAAIAAgACAAYgBvAGQAeQA6ACAAIgBIAGUAbABsAG8ALAAgAHcAbwByAGwAZAAhACIACgAgACAAIAAgAH0AOwAKACAAIAB9AAoAfQAKAAoAYQBwAGkALgBnAGUAdAAoACIALwAiACwAIABSAGUAcQB1AGUAcwB0AEMAbwBuAHQAcgBvAGwAbABlAHIALgBoAGEAbgBkAGwAZQBSAGUAcQB1AGUAcwB0ACkAOwA%3D)

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by ekeren** [comment]
There are actually two completely different issues here:
- [ ] inflight static method should be a first class citizen 
(more of a feature)
- [ ] TypeError: Cannot read properties of undefined
(seems like a regression issue)

**Comment by staycoolcall911** [comment]
The original issue showed 2 different issues #3251, #3252. After #3252 was closed as a duplicate of #457, we decided to close #3251 and keep this issue.
I revised the issue's title and description to reflect all that.

**Comment by eladb** [comment]
Updated title

**Comment by staycoolcall911** [comment]
Checked it now and the current status is quite the same:

Ran this:
```ts
bring cloud;

let api = new cloud.Api();

inflight class RequestController {
  static inflight handleRequest(request: cloud.ApiRequest): cloud.ApiResponse {
    return cloud.ApiResponse {
      status: 200,
      body: "Hello, world!"
    };
  }
}

api.get("/", RequestController.handleRequest);
```

Received this error:
```sh
error: Cannot reference an inflight value from within a preflight expression
   --> bla.w:14:14
   |
14 | api.get("/", RequestController.handleRequest);
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot reference an inflight value from within a preflight expression
```
